### PR TITLE
BCDA 4352: Add CLI command to remove data from cancelled jobs

### PR DESCRIPTION
**Pull request by katyasoup**
### Fixes [BCDA-4352](https://jira.cms.gov/browse/BCDA-4352)
In BCDA-[3660](https://github.com/CMSgov/bcda-app/pull/636)/[3661](https://github.com/CMSgov/bcda-app/pull/636) we introduced the ability to cancel in-progress jobs. Similar to [BCDA-3879](https://github.com/CMSgov/bcda-app/pull/621), we need a way to remove the stale data associated with a cancelled job.

### Change Details
- Add new cli command `cleanup-cancelled` that will remove data in the PAYLOAD and STAGING dirs.
- Introduce new job status, `CancelledExpired`, that will indicate that the data associated with the cancelled job is returned.

### Security Implications

- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [x] new data stored or transmitted
_When executed, it will clean up any data associated with a cancelled job._
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

### Acceptance Validation
- Unit tests updated and passing locally
- Ran updated Jenkins job to verify the process works on the dev environment
Screen Shot 2021-03-26 at 11 50 23 AM

### Feedback Requested
General feedback and improvements, better consolidation of tests

### CODE REVIEW

**Review by adamayd — APPROVED** [review]
LGTM!!

**Review by SMLuthi — APPROVED** [review]
LGTM!

File: bcda/bcdacli/cli_test.go [hunk 1]
```diff
@@ -489,6 +489,58 @@ func (s *CLITestSuite) TestCleanupFailed() {
...
+		for _, f := range append(shouldExist, shouldNotExist...) {
+			os.Remove(f.Name())
+			f.Close()
+		}
+	}()
+
+	err := s.testApp.Run([]string{"bcda", "cleanup-cancelled", "--threshold", strconv.Itoa(threshold)})
+	assert.NoError(s.T(), err)
+
+	assert.Equal(s.T(), models.JobStatusCancelledExpired,
```
[the remark this answers is not in the document]
Comment (katyasoup) [reply bcda/bcdacli/cli_test.go:522]: Can you elaborate? Not sure how else we test the cleanup functionality with the new status

[the remark this answers is not in the document]
Comment (CosmicSyntax) [reply bcda/bcdacli/cli_test.go:522]: Didn't see JobStatusCancelled -> JobStatusCancelledExpired

[the remark this answers is not in the document]
Comment (katyasoup) [reply bcda/bcdacli/cli_test.go:522]: Resolved offline :)

**Pull request merged**